src/libcore/slice/mod.rs

```diff
@@ -1362,6 +1362,23 @@ impl<T> [T] {
     /// let r = s.binary_search(&1);
     /// assert!(match r { Ok(1..=4) => true, _ => false, });
     /// ```
+    ///
+    /// Inserts a new element into already sorted array while maintaining sorted
+    /// order. Then asserts that the item was inserted correctly.
+    ///
+    /// ```
+    /// let mut s = vec![0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55];
+    /// let num = 14;
+    ///
+    /// match s.binary_search(&num) {
+    ///     Ok(_) => {}
```

Comment:
Hm, I think we should be inserting regardless of finding the item here, so filling out this `Ok` branch with an insert as well would be good, perhaps even illustrating that by doing something like `Ok(idx) | Err(idx) => s.insert(idx, num)` in an if let.

Reply:
I believe my new commit is more what you were looking for. Do I need to squash the commits or will the merge do that for me?

Reply:
You'll need to squash the commits; it also looks like you have a build failure here -- I suspect the `;` after the if let.